From: Ubuntu Desktop Team <ubuntu-desktop@lists.ubuntu.com>
Date: Thu, 26 Jul 2018 16:19:06 +0100
Subject: correct_logout_action

# Description: let the logout key do what it's supposed to do
# Ubuntu: https://launchpad.net/bugs/961501
# Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=671979
# Description: let the logout key do what it's supposed to do
# Ubuntu: https://launchpad.net/bugs/961501
# Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=671979
---
 plugins/media-keys/gsd-media-keys-manager.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/plugins/media-keys/gsd-media-keys-manager.c b/plugins/media-keys/gsd-media-keys-manager.c
index f12a39b..ba527e3 100644
--- a/plugins/media-keys/gsd-media-keys-manager.c
+++ b/plugins/media-keys/gsd-media-keys-manager.c
@@ -1135,6 +1135,12 @@ gnome_session_shutdown (GsdMediaKeysManager *manager)
         g_object_unref (proxy);
 }
 
+static void
+do_logout_action (GsdMediaKeysManager *manager)
+{
+        execute (manager, "gnome-session-quit --logout", FALSE);
+}
+
 static void
 do_eject_action_cb (GDrive              *drive,
                     GAsyncResult        *res,
@@ -2634,7 +2640,7 @@ do_action (GsdMediaKeysManager *manager,
                                  SOUND_ACTION_FLAG_IS_OUTPUT | SOUND_ACTION_FLAG_IS_PRECISE);
                 break;
         case LOGOUT_KEY:
-                gnome_session_shutdown (manager);
+                do_logout_action (manager);
                 break;
         case EJECT_KEY:
                 do_eject_action (manager);
